pkg/cri/store/image/image.go

```diff
@@ -50,22 +55,39 @@ type Image struct {
 	Pinned bool
 }
 
+// InfoProvider provides both content and info about content
+type InfoProvider interface {
+	content.Provider
+	Info(ctx context.Context, dgst digest.Digest) (content.Info, error)
+}
+
 // Store stores all images.
 type Store struct {
 	lock sync.RWMutex
 	// refCache is a containerd image reference to image id cache.
 	refCache map[string]string
-	// client is the containerd client.
-	client *containerd.Client
+
+	// images is the local image store
+	images images.Store
+
+	// content provider
+	provider InfoProvider
+
+	// platform represents the currently supported platform for images
+	// TODO: Make this store multi-platform
+	platform platforms.MatchComparer
+
 	// store is the internal image store indexed by image id.
 	store *store
 }
 
 // NewStore creates an image store.
-func NewStore(client *containerd.Client) *Store {
+func NewStore(img images.Store, provider InfoProvider, platform platforms.MatchComparer) *Store {
```

Comment:
Do we expect the list of options here to grow in future? (Otherwise we could consider defining some "options" struct that can be passed)

Reply:
No, I would prefer this actually replaced completely by the existing image store or a new common implementation which could support faster lookup and more data caching.

Reply:
Alright! 👍